CameraInspector/PhotoMaker.cpp

```diff
@@ -0,0 +1,60 @@
+#include "PhotoMaker.h"
+
+namespace CameraInspector
+{
+
+PhotoMaker::PhotoMaker() : saver_(nullptr), curr_frame_{}
+{
+}
+
+PhotoMaker::PhotoMaker(std::shared_ptr<IFrameSaver>& saver) : saver_(saver), curr_frame_{}
+{
+}
+
+void PhotoMaker::ProcessFrame(cv::Mat frame)
+{
+	curr_frame_ = frame;
+}
+
+void PhotoMaker::MakePhoto() const
+{
+	if (saver_)
+	{
+		saver_->SaveFrame(curr_frame_);
```

Comment:
So, if saver member is required to proper work of this class it make sense to implement constructor which initialize the saver.